openqaoa/optimizers/result.py

```diff
@@ -156,6 +157,90 @@ def plot_cost(
 
         return
 
+    def plot_probabilities(self, states_to_keep = None, figsize = (10,8),label='Probability distribution',color='tab:blue', ax=None):
+
+        """
+        Helper function to plot the probabilities corresponding to each basis states (with prob != 0) obtained from the optimized result
+
+        Parameters
+        ----------
+        states_to_keep: 'int
+            If the user passes a value, the plot will compile with the given value of states. Else, a max of 
+        figsize: `tuple`
+            The size of the figure to be plotted. Defaults to (10,8).
+        label: `str`
+            The label of the cost line, defaults to 'Probability distribution'.
+        color: `str`
+            The color of the line. Defaults to 'tab:blue'.
+        ax: 'matplotlib.axes._subplots.AxesSubplot'
+            Axis on which to plot the graph. Deafults to None
+        """
+
+        outcome = self.optimized['optimized measurement outcomes']
+
+        # converting to counts dictionary if outcome is statevector
+        if type(outcome) == type(np.array([])):
+            outcome = self.get_counts(outcome)
+            # setting norm to 1 since it might differ slightly for statevectors due to numerical preicision
+            norm = np.float64(1)
+        else: 
+            # needed to be able to divide the tuple by 'norm'
+            norm = np.float64(sum(outcome.values())) 
+
+        # sorting dictionary. adding a callback function to sort by values instead of keys
+        # setting reverse = True to be able to obtain the states with highest counts
+        outcome_list = sorted(outcome.items(), key=lambda item: item[1], reverse=True)
+        states, counts = zip(*outcome_list)
+
+        # normalizing to obtain probabilities
+        probs = counts/norm
+
+        # total number of states / number of states with != 0 counts for shot simulators
+        total = len(states)
+
+        # number of states that fit without distortion in figure
+        upper_bound = 40
+        # default fontsize
+        font = 'medium'
+
+        if states_to_keep:
+            if states_to_keep > total:
+                raise TypeError('states_to_keep must be smaller than the total number of states')
```

Comment:
I think it needs to be a ValueError, and it could be useful to have a reference of what the total is

replace to `raise ValueError(f'states_to_keep must be smaller or equal to the total number of states {total}')`

And should it say `smaller or equal`, right?

Reply:
Thank you for the comments! I implemented all of them, the only difference is that I included 'number of states in measurement outcome' in the exception. This is to make it clear because sometimes the total states differ from 2^qubits when doing shots simulators and some states have zero counts